### PR TITLE
Fix implementation of collection snapshot

### PR DESCRIPTION
**Pull request by leplatrem**
The previous implementation was not handling properly deletions/recreations. This one is much simpler and works as expected.

### CODE REVIEW

File: src/collection.ts [hunk 1]
```diff
@@ -780,13 +780,14 @@ export default class Collection {
...
       pages: Infinity, // all pages up to target timestamp are required
-      sort: "-target.data.last_modified",
+      sort: "target.data.last_modified",
       filters: {
         resource_name: "record",
         collection_id: this.name,
         "max_target.data.last_modified": String(at), // eq. to <=
       },
     });
+    console.log(`${changes.length} history entries found.`);
```
Comment (dstaley) [inline src/collection.ts:790]: We should probably figure out a system to allow for optional debugging statements, but for now let's remove calls to `console.log` and `console.debug`.

**Review by dstaley — CHANGES_REQUESTED** [review]
Looks great! Definitely easier to read.

File: src/collection.ts [hunk 2]
```diff
@@ -801,30 +801,30 @@ export default class Collection {
...
-        snapshot = snapshot.filter((r) => r.id !== record.id);
-      } else if (!seenIds.has(record.id)) {
-        seenIds.add(record.id);
-        snapshot.push(record);
+        recordsById.delete(record.id);
+      } else {
+        recordsById.set(record.id, record);
       }
     }
+    console.debug(`Found ${recordsById.size} records in snapshot (at=${at}).`);
```
Comment (dstaley) [inline src/collection.ts:817]: Should we keep this?

**Review by dstaley — APPROVED** [review]
I'll leave it up to you if you want to remove the call to `console.debug`.

File: src/collection.ts (hunk 2 above)
Comment (leplatrem) [reply src/collection.ts]: No, I agree, this shouldn't be the first one

**Pull request merged**